tests/phpunit/Mutator/Number/DecrementIntegerTest.php

```diff
@@ -519,5 +519,24 @@ public function mutationsProvider(): iterable
 }
 PHP
         ];
+
+        $minIntPlus1 = PHP_INT_MIN + 1;
+
+        yield 'It will decrement min int plus one' => [
+            <<<"PHP"
+            <?php
+
+            if (1 === {$minIntPlus1}) {
+                echo 'bar';
+            }
+            PHP,
+            <<<"PHP"
+            <?php
+
+            if (1 === -({$minIntPlus1}-1)) {
```

Comment:
That's what PHP Parser produces. I think I mentioned this in the description, can you see?

Reply:
Sorry, I think I clearly understood the issue from the beginning, but this line of code in tests is very very confusing.

Let's talk with numbers, not variable names. 

`PHP_MIN_INT` is `-9223372036854775808`
`$minIntPlus1` is `-9223372036854775807`

So, if we forget about PHP Parser bug, then decrementing `-9223372036854775807` must be `PHP_INT_MIN` which is `-9223372036854775808`, right? Right.

Then let's get back to PHP Parser bug. We know that the result should be integer `-9223372036854775808`, but this can not be correctly parsed by PHP and it is converted to `float(9.223372036854776E+18)`. To workaround it, we added casting to int, which brings `float(9.223372036854776E+18)` back to `int(-9223372036854775808)`.

But where `-({$minIntPlus1}-1)` comes from? As I understand, when PHP Parser sees `int(-9223372036854775808)`, it replaces it with `-9223372036854775807 - 1` expression **automatically** (we don't do it anywhere in our mutators). 

So we have `-(-9223372036854775807-1)` which is `-(-9223372036854775808)` which is `+9223372036854775808` which in its turn [leads to INT overflow](https://3v4l.org/X25ao) and becomes `-9223372036854775808`

Is it how you see it as well? (above is based on my results with debugger, stepping through `DecrementInteger` code

Reply:
Do you mean we shouldn't allow to decrement PHP_MIN_INT + 1?

Reply:
The more I think about it the more confusing it gets. Why do we increment `PHP_MIN_INT + 1` in in a mutator named `DecrementInteger`?

Reply:
Let's circle back a little. We're discussing a test for `DecrementInteger`. In this PR it had only one change: added integer cast. So if we're seeing something funny here, it must be another PHP Parser bug. What do you think we should do?

Reply:
I guess we can tell both of these mutators to back off if they see PHP_MIN_MAX. We'll have a slightly dysfunctional mutator but we'll avoid these bugs. If we consider that there's no workaround, this seem like a logical choice.